src/components/authentication-error-dialog/index.tsx

```diff
@@ -0,0 +1,30 @@
+import useAuthentication from 'hooks/use-authentication'
+import Button from 'components/button'
+import Dialog from 'components/dialog'
+import Text from 'components/text'
+import s from './authentication-error-dialog.module.css'
+import { IconSignIn16 } from '@hashicorp/flight-icons/svg-react/sign-in-16'
+
+const AuthenticationErrorDialog = () => {
+	const { hasError, signIn } = useAuthentication()
+	const label = 'Are you still there?'
+
+	return (
+		<Dialog isDismissable={false} isOpen={hasError} label={label}>
+			<Text asElement="p" size={300} weight="semibold">
+				{label}
+			</Text>
+			<Text asElement="p" className={s.body}>
+				Your session has expired due to inactivity. Please sign in to continue.
+			</Text>
+			<Button
+				icon={<IconSignIn16 />}
+				iconPosition="leading"
+				text="Sign in"
+				onClick={() => signIn()}
```

Comment:
💭 I think we should provide an option if a user doesn't want to sign back in. Majority of our site doesn't require authentication to interact with it. Could we provide an 'x' to dismiss (and sign out the user, or just leave it) or should we have an explicit button, stating an option for not signing in?

Reply:
Having two separate buttons was in the original proposal, but design decided to go with one to keep things simplified. I don't think we'd want the `X` to do anything automatically because the user won't know that's what happened. @kellyschummer do you have any additional context on not having a Sign out button present?

Reply:
🗒️ PR has been updated and this convo is no longer relevant, so marking resolved.